### ISSUE TITLE
Nvda does not return with the exploration mode after closing save as in chrome window

### ISSUE THREAD

**Issue by fernando-jose-silva** [nvaccess/nvda#7099]
To reproduce:
Using the chrome open a page where there is a file to be downloaded as an example the nvda page:
Https://www.nvaccess.org/files/nvda/snapshots/
Now on this page, move the arrows to a download link.
Press the application (right mouse button), and open the option:
"Save this link as"
Now without saving the file, press esc to return to the web page.
Expected when pressing arrows to any side the nvda should read what's on the web page.
What happens, when you press any arrow the nvda does not read anything.
It is necessary to press as tabe example to be able to navigate again with the arrows by the page.

**Comment by ehollig** [comment]
Confirmed using Chrome 60 and the steps provided in the original post. Pressing tab however, returns everything back to normal.

**Comment by Adriani90** [comment]
I can reproduce this issue in Firefox 66.0.3 and Chrome 73 with NVDA 2019.1.
Following is in the log:

```
IO - inputCore.InputManager.executeGesture (00:21:45.711):
Input: kb(laptop):escape
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (00:21:45.841):
oleacc.AccessibleObjectFromEvent with window 2295818, objectID -4 and childID 0: [Error -2147024809] Falscher Parameter
IO - inputCore.logTimeSinceInput (00:21:45.891):
0.179 sec since input
IO - speech.speak (00:21:45.891):
Speaking [u'google - Google-Suche - Google Chrome  nicht verf\xfcgbar']
IO - inputCore.InputManager.executeGesture (00:21:48.921):
Input: kb(laptop):tab
DEBUGWARNING - NVDAObjects.IAccessible.IAccessible._get_IAccessibleRole (00:21:48.960):
accRole failed: (-2147024809, 'Falscher Parameter.', (None, None, None, 0, None))
IO - inputCore.logTimeSinceInput (00:21:49.040):
0.120 sec since input
IO - speech.speak (00:21:49.040):
Speaking [u'google - Google-Suche  Dokument']
```

**Comment by Adriani90** [comment]
related to #8389

**Comment by Adriani90** [comment]
Hmm, I could catch following error in Chrome 79 while reproducing this issue with NVDA 2019.3 Beta 1:

```
DEBUG - appModuleHandler.cleanup (16:54:07.914) - MainThread (10016):
application chrome closed
IO - inputCore.logTimeSinceInput (16:54:07.924) - MainThread (10016):
0.840 sec since input
IO - speech.speak (16:54:07.924) - MainThread (10016):
Speaking ['NVDA Development Snapshots - Google Chrome']
IO - inputCore.InputManager.executeGesture (16:54:08.484) - winInputHook (12968):
Input: kb(laptop):upArrow
IO - inputCore.InputManager.executeGesture (16:54:09.010) - winInputHook (12968):
Input: kb(laptop):downArrow
IO - inputCore.InputManager.executeGesture (16:54:09.395) - winInputHook (12968):
Input: kb(laptop):downArrow
IO - inputCore.InputManager.executeGesture (16:54:09.713) - winInputHook (12968):
Input: kb(laptop):upArrow
IO - inputCore.InputManager.executeGesture (16:54:10.045) - winInputHook (12968):
Input: kb(laptop):upArrow
IO - inputCore.InputManager.executeGesture (16:54:10.254) - winInputHook (12968):
Input: kb(laptop):downArrow
IO - inputCore.InputManager.executeGesture (16:54:10.468) - winInputHook (12968):
Input: kb(laptop):downArrow
IO - inputCore.InputManager.executeGesture (16:54:10.712) - winInputHook (12968):
Input: kb(laptop):upArrow
IO - inputCore.InputManager.executeGesture (16:54:10.931) - winInputHook (12968):
Input: kb(laptop):downArrow
IO - inputCore.InputManager.executeGesture (16:54:11.472) - winInputHook (12968):
Input: kb(laptop):tab
IO - inputCore.logTimeSinceInput (16:54:11.648) - MainThread (10016):
0.176 sec since input
IO - speech.speak (16:54:11.648) - MainThread (10016):
Speaking ['NVDA Development Snapshots', 'Dokument']
IO - inputCore.logTimeSinceInput (16:54:11.654) - MainThread (10016):
0.182 sec since input
IO - speech.speak (16:54:11.654) - MainThread (10016):
Speaking ['Link', 'alpha-19441,855bf207']
IO - inputCore.logTimeSinceInput (16:54:11.660) - MainThread (10016):
0.188 sec since input
IO - speech.speak (16:54:11.660) - MainThread (10016):
Speaking ['Download', 'Link', 'alpha-19441,855bf207']
```

**Comment by Adriani90** [comment]
It seems after pressing escape to close the save as dialog, chrome sugerates that it closed the whole application which is not true. The application is still open and regains focus when pressing tab.
I don't know if this is a bug in the Chrome app module in NVDA or in Chrome itself.

cc: @feerrenrut, @michaelDCurran

**Comment by Adriani90** [comment]
I guess in Firefox happens the same but there is nothing written to the log.
cc: @jcsteh. 
It might be also related to #8389.

**Comment by jcsteh** [comment]
For firefox, NVDA was tracking this in #2497 and now #10450. The Mozilla bug is https://bugzilla.mozilla.org/show_bug.cgi?id=782547. The good news is that there's been some progress on this lately, but the fix hasn't landed yet.

Note that this will need to be fixed separately in Chrome. I'm not sure if they have a bug on file or not.

**Comment by jcsteh** [comment]
This is (finally!) fixed for Firefox 74 (currently available for testing in Firefox Nightly).

**Comment by Adriani90** [comment]
Still reproducible with the save as dialog (ctrl+s) both on Firefox 74 and Chrome 80 with NVDA Alpha 19774,f962301d

**Comment by jcsteh** [comment]
It turned out that while the Print dialog was fixed for Firefox 74, [an additional fix was needed for the Save and Open dialogs](https://bugzilla.mozilla.org/show_bug.cgi?id=1606932). That didn't make it into Firefox 74, but is available in Firefox 75 (currently in beta).

**Comment by Adriani90** [comment]
This is fixed in Firefox 75 indeed, thanks for that Jamie.

This issue exists now only in Crome, also tested in Crome 81.

**Comment by ObjectInSpace** [comment]
This is tracked in crbug/995551 on the Chrome side.

**Comment by Adriani90** [comment]
Fixed in Chrome, I tested with NVDA alpha-53273,a6a536eb (2026.1.0.53273) and it works as expected. Closing as works for me.
